text/0000-optional-jquery.md

```diff
@@ -0,0 +1,248 @@
+
+### But this is already possible, why this RFC?
+
+There is indeed a somewhat quirky way to build an [app without jQuery](https://github.com/rwjblue/no-jquery-app) even
+today.
+Although this *happens* to work, it is not sufficient to consider this officially supported for these reasons:
+* Ember itself must be fully tested to work without jQuery
+* the public APIs that depend on and/or expose jQuery need to have some well defined behavior when jQuery is not 
+available
+* there should be a way to technically opt-out (other than fiddling with [`vendorFiles`](https://github.com/rwjblue/no-jquery-app/commit/34c40fc2cfc5e2ce0c39e5e906448c46af699d26))
+that is easier to use, understand and maintain
+* addons should mostly default to not use jQuery, to make removing jQuery practically possible for their consuming
+apps
+
+## Detailed design
+
+### Remove internal jQuery usage
+
+As of writing this, there are [major efforts](https://github.com/emberjs/ember.js/issues/16058) underway to remove and 
+cleanup the Ember codebase and especially its tests from jQuery usage. Having a way to fully test Ember without jQuery 
+is a prerequisite to officially support jQuery being optional. When this is done, it will enable a "no jQuery" mode, 
+that will make it not use jQuery anymore, but only native DOM APIs.
+
+### Add an opt-out flag
+
+There should be a global flag that will toggle the optional jQuery integration (true by default). When this is disabled,
+it will make Ember CLI's build process *not* include jQuery into the `vendor.js` bundle, *and* it will explicitly put 
+Ember itself into its "no jQuery" mode.
+
+The flag itself will not be made a public API. Rather it will be handled by a privileged addon, that will allow to 
+disable the integration flag, thus to opt out from jQuery integration. This approach is in line with 
+[RFC 278](https://github.com/emberjs/rfcs/pull/278) and [RFC 280](https://github.com/emberjs/rfcs/pull/280), to allow 
+for some better implementation flexibility.
+
+### Introduce `@ember/jquery` package
+
+Currently Ember CLI itself is importing jQuery into the app's `vendor.js` file. To decouple it from this task, and
+to allow for some better flexibility in the future, the responsibility for importing jQuery is moved to a dedicated 
+`@ember/jquery` addon.
+
+To not create any breaking changes, Ember CLI will have to check the app's dependencies for the presence of this addon. 
+If it is not present, it will continue importing jQuery *unless* the jQuery integration flag is disabled.
+In the other case, it will stop importing jQuery at all, and delegate this responsibility to the addon.
+
+To nudge users to install `@ember/jquery` when they need jQuery, some warning/deprecation messages should be issued when
+the addon is *not* installed and the integration flag is either not specified or is set to true. Only in the case the 
+app is actively opting out of jQuery integration the addon is not needed.
```

Comment:
The `@ember/jquery` addon should be placed in the default blueprint for a period of time so as to ease migration.

Reply:
Yep, absolutely agreed.